### PR TITLE
Add TSVECTOR and match operator (@@) for PostgreSQL

### LINKED ISSUE #1580: Fulltext support
Adding support for this would be useful as it's becoming quite popular as a simpler alternative to systems like Solr, Sphinx and ElasticSearch.

Here are the docs for text search:
http://www.postgresql.org/docs/9.3/static/textsearch.html

Also, I don't know how useful this is, but here is a nice Ruby gem for Postgres full text which may help with laying out a plan on how to do it with sequelize.
https://github.com/Casecommons/pg_search

### PR DESCRIPTION
**Pull request by janokacer**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Added TSVECTOR datatype and @@ operator for postgresql more information about datatype here:
https://www.postgresql.org/docs/current/functions-textsearch.html

Closes #1580

### CODE REVIEW

**Comment by jnelken** [comment]
Here is the fulltext issue related to this PR https://github.com/sequelize/sequelize/issues/1580

File: docs/models-usage.md [hunk 1]
```diff
@@ -193,6 +193,7 @@ Project.findAll({
       [Op.contains]: [1, 2],      // @> [1, 2] (PG array contains operator)
       [Op.contained]: [1, 2],     // <@ [1, 2] (PG array contained by operator)
       [Op.any]: [2,3]            // ANY ARRAY[2, 3]::INTEGER (PG only)
+      [Op.match]: Sequelize.fn('to_tsquery', 'fat & rat') // math text search for strings 'fat' and 'rat'
```
Comment (felamaslen) [inline docs/models-usage.md]: Typo in "math", should be "match"

**Comment by papb** [comment]
@janokacer Hello, this PR has conflicts, can you please rebase?

**Comment by janokacer** [comment]
> 
> 
> @janokacer Hello, this PR has conflicts, can you please rebase?

Hello @papb, I just did. I hope it helps.

**Comment by papb** [comment]
Fantastic, thank you very much. I will review now

**Comment by papb** [comment]
Hello, anyone that is reading this, would be willing to continue the work on this PR? You can fork OP's fork and continue from there. Thanks!

**Comment by cauefernandes** [comment]
@papb what's the issue here preventing merge?

**Comment by djake** [comment]
I am currently writing the SQL by hand due to lack of support for the match operator, which stinks because I lose all the Sequelize goodness. 

@cauefernandes (or anyone else) are you planning on bringing this to completion soon? If not I'm more than happy to do so.

**Comment by djake** [comment]
Closing this in favor of #12631 which has an integration test a fewer conflicts. Thank you for this.

**Pull request closed without merging**